docs/graph_engine/custom_decoder.rst

```diff
@@ -0,0 +1,130 @@
+**************************************************
+How to Create a Custom Decoder from Avro to Binary
+**************************************************
+
+In this example we take an Avro file as input, write a Avro to binary
+decoder and then use this decoder in to generate binaries
+that can be loaded by our graph engine.
+
+First we generate an example Avro file to use. This is a binary format with a schema to
+define the columns and their types and a binary file where each row is a record.
+This table has the columns: "src", "dst", "type", "weight", "float_feature" and "int_feature".
+Each row represents one record, either a node or an edge.
+
+In this file we have two nodes, 0 and 1, on lines 0 and 2 in which the src column gives their
+node ID and the dst is -1. We also have an edge in each direction between the two on lines 1 and 3.
+It is important that we order these rows specifically, first with a node then all of its outgoing
+edges in order of type then dst, then the next node and so on.
+Each of these items have four fields: a type, weight, float feature vector and integer feature vector.
+For any graph you use, all items are required to have a type and weight, then they can have a variable
+number of indexable feature vectors.
+
+.. code-block:: python
+
+    >>> import tempfile
+    >>> import avro.schema
+    >>> from avro.datafile import DataFileReader, DataFileWriter
+    >>> from avro.io import DatumReader, DatumWriter
+    >>> working_dir = tempfile.TemporaryDirectory()
+    >>> raw_file = working_dir.name + "/data.avro"
+
+    >>> schema = avro.schema.parse("""{"namespace": "example.avro", "type": "record", "name": "Graph", "fields": [
+    ... {"name": "src", "type": "int"},
+    ... {"name": "dst",  "type": "int"},
+    ... {"name": "type", "type": "int"},
+    ... {"name": "weight", "type": "float"},
+    ... {"name": "float_feature", "type": ["bytes", "null"]},
```

Comment:
why do we use type `bytes` for float features? This is confusing.

Reply:
Bytes is the way you store arrays in avro, its just np.tobytes() and np.frombuffer()

Reply:
Of course not: https://avro.apache.org/docs/1.11.1/specification/#arrays